Peer.Domain/Commands/Show.cs

```diff
@@ -16,30 +20,66 @@ public class Show
         private readonly ISorter<PullRequest>? _sorter;
         private readonly List<IFilter> _filters;
 
+        public ShowConfig Config { get; }
+
         public Show(
             IPullRequestService prService,
             IListFormatter formatter,
             IConsoleWriter writer,
+            ShowConfig config,
             ISorter<PullRequest>? sorter = null,
             IEnumerable<IFilter>? filters = null)
         {
             _pullRequestService = prService;
             _formatter = formatter;
             _writer = writer;
+            Config = config;
             _sorter = sorter;
             _filters = filters?.ToList() ?? new();
         }
 
         public async Task<Result<None, ShowError>> ShowAsync(ShowArguments args, CancellationToken token = default)
         {
-            var prs = await _pullRequestService.FetchAllPullRequests(token);
-            prs = _filters.Aggregate(prs, (prs, filter) => filter.Filter(prs));
+            var prs = await GetPullRequests(args, token);
+            if (prs.IsError)
+            {
+                _writer.Clear();
+                _writer.Display(new List<string>
+                {
+                    $"error: failed to fetch pull request info",
+                }, token);
+                return ShowError.Fire;
+            }
 
-            var sorted = await (_sorter?.Sort(prs) ?? prs).Take(args.Count).ToListAsync(token);
-            var lines = _formatter.FormatLines(sorted).ToList();
+            var lines = _formatter.FormatLines(prs.Value).ToList();
             _writer.Display(lines, token);
             return Maybe.None;
         }
+
+        private async Task<Result<IList<PullRequest>, ShowError>> GetPullRequests(ShowArguments args, CancellationToken token)
+        {
+            var retryNumber = 0;
+            var stopWatch = Stopwatch.StartNew();
+
+            while (true)
+            {
+                try
+                {
+                    var prs = await _pullRequestService.FetchAllPullRequests(token);
```

Comment:
@Insomniak47  you had mentioned having retry in the fetchers and that seems better in that it would only re-request the one(s) that failed, but where we've used IAsyncEnumerable I'm not sure how we'd handle it would work. Specifically, I don't know how we would instigate the retry in a case where we may have already yielded some elements in the enumerable like if it was a next-page request that failed.

Reply:
We run .Distinct on the prs and we're yielding only the prs back from the async enumerables so (at least on the github side) we're always just enumerating everything and then yielding back rn so I don't think this ends up being a problem does it? Each individual query is realized and then yielded and if we were paging github prs we could .Distinct on the consumption side of things to prevent dupes between pages. We'd have to do something similar if we were using the graphql client.

I'm assuming we're not going to retry for things like errors realizing PR objects (nulls etc) so I think we can just factor out the retry logic into a RunQueryAsync function in the github fetcher that handles it. 

I'm pretty tired so I might be missing something though

Reply:
We do single full pages of PRs, but within a page of PRs we do [this](https://github.com/wareismymind/peer/blob/01f5d1f4bf4c781d34be47b779443a18c2ae2723/Peer.GitHub/GitHubRequestFetcher.cs#L68)

```
while (prsWithMoreThreads.Any())
{
    var queryResponse = await _gqlClient.SendQueryAsync<Dictionary<string, PRSearch.PullRequest>>(
        ThreadPageQuery(prsWithMoreThreads),
        token);
    ...
```
so in that case we may have yielded some PRs before we get a failure on subsequent pages of threads.

We could wait to yield any PRs until all their thread pages are ready though, and we could retry each page request individually. It would make the retry count kind of weird but we always just the use retry cutoff and not worry about how many requests that turns out to be.